src/bodywork/git.py

```diff
@@ -44,102 +50,162 @@ def download_project_code_from_repo(
-            msg = (f'failed to setup SSH for GitHub - cannot find '
-                   f'{SSH_GITHUB_KEY_ENV_VAR} environment variable')
+        if SSH_PRIVATE_KEY_ENV_VAR not in os.environ:
+            msg = (
+                f"failed to setup SSH for {hostname} - cannot find "
+                f"{SSH_PRIVATE_KEY_ENV_VAR} environment variable"
+            )
             raise KeyError(msg)
-        ssh_dir.mkdir(mode=0o700, exist_ok=True)
-        private_key.touch(0o700, exist_ok=False)
-        private_key.write_text(os.environ[SSH_GITHUB_KEY_ENV_VAR])
-
-    known_hosts = ssh_dir / 'known_hosts'
-    if not known_hosts.exists():
-        ssh_public_key = 'github.com ssh-rsa AAAAB3NzaC1yc2EAAAABIwAAAQEAq2A7hRGmdnm9tUDbO9IDSwBK6TbQa+PXYPCPy6rbTrTtw7PHkccKrpp0yVhp5HdEIcKr6pLlVDBfOLX9QUsyCOV0wzfjIJNlGEYsdlLJizHhbn2mUjvSAHQqZETYP81eFzLQNnPHt4EVVUh7VfDESU84KezmD5QlWpXLmvU31/yMf+Se8xhHTvKSCZIFImWwoG6mbUoWf9nzpIoaSjB+weqqUUmpaaasXVal72J+UX2B+2RPW3RcT0eOzQgqlJL3RKrTJvdsjE3JEAvGq3lGHSZXy28G3skua2SmVi/w4yCE6gbODqnTWlg7+wC604ydGXA8VJiS5ap43JXiUFFAaQ=='  # noqa
-        known_hosts.touch(0o700, exist_ok=False)
-        known_hosts.write_text(ssh_public_key)
-
-    os.environ['GIT_SSH_COMMAND'] = (
-        f'ssh -i {private_key} -o UserKnownHostsFile={known_hosts}'
+        try:
+            ssh_dir.mkdir(mode=0o700, exist_ok=True)
+            private_key.touch(0o700, exist_ok=False)
+            private_key.write_text(os.environ[SSH_PRIVATE_KEY_ENV_VAR])
+        except OSError as e:
+            raise RuntimeError(
+                f"Unable to create private key {private_key} from"
+                f" {SSH_PRIVATE_KEY_ENV_VAR} environment variable"
+            ) from e
+
+    try:
+        known_hosts = ssh_dir / "known_hosts"
+        if not known_hosts.exists():
+            known_hosts.touch(0o700, exist_ok=False)
+            known_hosts.write_text(get_ssh_public_key_from_domain(hostname))
+        elif not known_hosts_contains_domain_key(hostname, known_hosts):
+            known_hosts.touch(0o700)
+            with known_hosts.open(mode="a") as file_handle:
+                file_handle.write(get_ssh_public_key_from_domain(hostname))
+    except OSError as e:
+        raise RuntimeError(
+            f"Error updating known hosts with public key from {hostname}"
+        ) from e
+
+    os.environ["GIT_SSH_COMMAND"] = (
+        f"ssh -i '{private_key}'" f" -o UserKnownHostsFile='{known_hosts}'"
     )
+
+
+def known_hosts_contains_domain_key(hostname: str, known_hosts_filepath: Path) -> bool:
+    """Checks to see if the host is in the list of keys in the known_hosts file.
+
+    :param known_hosts_filepath: path to known_hosts file
+    :param hostname: Host name to check for.
+    :return: bool if the hostname is in the file
+    """
+    return hostname in known_hosts_filepath.read_text()
+
+
+def get_ssh_public_key_from_domain(hostname: str) -> str:
+    """Gets the public SSH key from the host and checks if the fingerprint matches
```

Comment:
Need to format the docstring (same for the above). I'll see if I can find a linter.

Reply:
Have added a space between the first line and the description otherwise I don't what the docstring requirements are.